### PR TITLE
scdoc: refactor

### PR DESCRIPTION
**Pull request by AndersonTorres**
- one parameter per line
- use rec-less, overlay-style overridable recursive attributes (in effect since https://github.com/NixOS/nixpkgs/pull/119942)
- remove references to pname
- SRI hash
- multiple outputs
- remove with nesting (following https://nix.dev/anti-patterns/language#with-attrset-expression)
- simplify meta
- add myself as maintainer

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Maybe it would generate a dreadful mass rebuild?

**Comment by AndersonTorres** [comment]
@ofborg build pkgsLLVM.yambar

**Comment by AndersonTorres** [comment]
@ofborg build pkgs.yambar

File: pkgs/tools/typesetting/scdoc/default.nix [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
-      --replace "-static" "" \
-      --replace "/usr/local" "$out"
+      --replace "-static" ""
   '';
 
-  makeFlags = lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
-    "HOST_SCDOC=${buildPackages.scdoc}/bin/scdoc"
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+  ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
```
Comment (amjoseph-nixpkgs) [inline pkgs/tools/typesetting/scdoc/default.nix:27]:
```suggestion
  ] ++ lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
```

**Review by ghost — APPROVED** [review]
Built on:
- [x] `x86_64-linux`
- [x] `powerpc64le-linux`
- [x] `aarch64-linux` (cross from `x86_64-linux`)
- [x] `mips64el-linux-gnuabi64` (cross from `x86_64-linux`)
- [x] `mips64el-linux-gnuabin32` (cross from `x86_64-linux`)

**Comment by AndersonTorres** [comment]
Small reminder: set ´meta.mainProgram´.

**Pull request merged**